src/mlpack/core/optimizers/saga/saga_impl.hpp

```diff
@@ -124,6 +124,7 @@ double SAGAType<UpdatePolicyType, DecayPolicyType>::Optimize(
     for (size_t f = 0, b=0, currentFunction = 0; f < numFunctions;
       /* incrementing done manually */)
     {
+      mlpack::math::RandomSeed(time(NULL));
```

Comment:
math::RandInt should return different values for each step. So this isn't needed, as @rcurtin pointed out we could restart the test, something like https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/recurrent_network_test.cpp#L83. let me know if I should clarify the point.

Reply:
You're trying to say that we should test that SAGA should succeed atleast one time but @zoq  I'm very much curious that why in the world setting just a RandomSeed() can help SAGA to converge on all the variants of `Rosenbrock function`

Reply:
``RandomSeed()`` doesn't help, see  https://travis-ci.org/mlpack/mlpack/builds/355241650?utm_source=github_status&utm_medium=notification each build failed.

Reply:
Ideally we could find more robust settings, but if we can't the solution above should work.

Reply:
@zoq I'm extremely sorry for the delay. I was not feeling well since yesterday :(